packages/gasket-nextjs/README.md

```diff
@@ -47,41 +50,101 @@ However, there may be situations where you want to inject the `gasketData`
-  static async getInitialProps(ctx) {
-    const initialProps = await Document.getInitialProps(ctx)
-    return { ...initialProps }
-  }
-
-  render() {
-    return (
-      <Html>
-        <Head />
-        <body>
-          <Main />
-          <footer>Some custom content</footer>
-          <NextScript />
-        </body>
-      </Html>
-    )
-  }
+    static async getInitialProps(ctx) {
+        const initialProps = await Document.getInitialProps(ctx)
+        return {...initialProps}
+    }
+
+    render() {
+        return (
+            <Html>
+                <Head/>
+                <body>
+                <Main/>
+                <footer>Some custom content</footer>
+                <NextScript/>
+                </body>
+            </Html>
+        )
+    }
 }
 
-export default withGasketData({ index: 2 })(MyDocument);
+export default withGasketData({index: 2})(MyDocument);
 ```
 
 In this example, the `gasketData` script will be injected after the custom
 `<footer/>` instead of right after `<Main/>`.
 
 This is especially useful if you are somehow nesting or extending the `<Main/>`
-and `<NextScript/>` components and the decorator cannot find the right place to
-inject the script.
+and `<NextScript/>` components and the decorator cannot find the right place to inject the script.
+
+
+---
+
+### withGasketDataProvider
+
+Use this to inject gasket data in to provider context to share the gasketData with the `useGasketData` hook. This is SSR
+and client side friendly and can be added at the app level or component level.
+
+#### Example
+
+---
+
+```jsx
+// pages/_app.js
+import { AppProps } from 'next/app';
+import { withGasketDataProvider } from '@gasket/nextjs';
+
+
+const Root = ({ Page, pageProps }) => {
+  return (
+    <Page {...pageProps} />
+  );
+};
+
+export default withGasketDataProvider()(Root);
+```
+
+### useGasketData
+
+Use this hook to access the gasketData provided by the `withGasketDataProvider` hoc.
+
+#### Example
+
+
+```jsx
+// MyComponent.js
+import { useGasketData } from '@gasket/nextjs';
+
+
+export const MyComponent = (props) => {
+  const gasketData = useGasketData();
+
+  return (
+    <>
+      <div>{gasketData.something}</div>
+      <div>{gasketData.here}</div>
+    </>
+  );
+};
+```
+
+## How it works
+
+During server side lifecycles' data can be added to the gasketData property. When the `withGasketData` hoc is added to a custom nextJs _document, the gasketData is added to the rendered html inside a script tag.
+
+The `withGasketDataProvider` can be added to a nextJS custom _app or react component. This HOC will capture the gasket data from server side gasketData property or the script tag that was rendered to the html. If nextJs is preforming a SSR the data will come from the gasketData property, otherwise the data will come from the script tag. The `withGasketDataProvider` hoc will then create a provider and inject gasket data into a context that can be consumed by the `useGasketData` hook.
```

Comment:
_nit_: keep consist "proper name" for Next.js
```suggestion
During server side lifecycles' data can be added to the gasketData property. When the `withGasketData` hoc is added to a custom Next.js _document, the gasketData is added to the rendered html inside a script tag.

The `withGasketDataProvider` can be added to a Next.js custom _app or react component. This HOC will capture the gasket data from server side gasketData property or the script tag that was rendered to the html. If Next.js is preforming a SSR the data will come from the gasketData property, otherwise the data will come from the script tag. The `withGasketDataProvider` hoc will then create a provider and inject gasket data into a context that can be consumed by the `useGasketData` hook.
```